src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -248,31 +272,47 @@ private void RemoveProcessedTransactions(Transaction[] blockTransactions)
 
             for (int i = 0; i < blockTransactions.Length; i++)
             {
-                Transaction transaction = blockTransactions[i];
-                Hash256 txHash = transaction.Hash ?? throw new ArgumentException("Hash was unexpectedly null!");
+                Transaction blockTx = blockTransactions[i];
+                Hash256 txHash = blockTx.Hash ?? throw new ArgumentException("Hash was unexpectedly null!");
 
-                if (!IsKnown(txHash))
+                if (blockTx.Supports1559)
                 {
-                    discoveredForHashCache++;
+                    eip1559Txs++;
                 }
 
-                if (!RemoveIncludedTransaction(transaction))
+                if (blockTx.SupportsBlobs)
                 {
-                    discoveredForPendingTxs++;
+                    blobTxs++;
+                    blobs += blockTx.BlobVersionedHashes?.Length ?? 0;
+
+                    if (_blobReorgsSupportEnabled)
+                    {
+                        if (_blobTransactions.TryGetValue(blockTx.Hash, out Transaction? fullBlobTx))
+                        {
+                            if (_logger.IsTrace) _logger.Trace($"Saved processed blob tx {blockTx.Hash} from block {block.Number} to ProcessedTxs db");
+                            blobTxsToSave ??= new List<Transaction>(Eip4844Constants.GetMaxBlobsPerBlock());
+                            blobTxsToSave.Add(fullBlobTx);
```

Comment:
So I think you could remove the need of "blobTxsToSave" list with saving to db and db batches

Reply:
If not you could easily change it to lazy evaluated `IEnumerable` or at least `ArrayPoolList`, but batch could be even simpler.

Reply:
It works this way - collecting txs, then creating one long RLP with all of them and then saving to db as one record, where block number is a key and all blob txs from this block (as RLP) are a value. So we need to collect all txs before encoding - otherwise we don't know message length which is a prerequisit for encoding. As we need to collect all txs at some point, lazy enumeration will just move this part to BlobTxStorage, but we will not avoid it. Doing batch write will not help here, as we are doing only one write, unless I am missing something. I just changed `blobTxsToSave` to be ArrayPoolList